packages/api/src/external/carequality/organization-template.ts

```diff
@@ -0,0 +1,211 @@
+  const endpointXCPD = getEndpoint(oid, urlXCPD, XCPD_STRING, "ihe-xcpd", "Patient Discovery");
+  const endpointDQ = getEndpoint(oid, urlDQ, XCA_DQ_STRING, "ihe-xca", "Query for Documents");
+  const endpointDR = getEndpoint(oid, urlDR, XCA_DR_STRING, "ihe-xca", "Retrieve Documents");
+
+  return `
+  <Organization>
+<!--This is the XCA/XCPD HomeCommunityID OID in URN format-->
+<identifier>
+    <use value="official"/>
+    <system value="http://www.hl7.org/oid/"/>
+    <value value="${oid}"/>
+</identifier>
+<active value="true"/>
+<name value="${orgName}"/>
+<type>
+    <coding>
+        <system value="http://hl7.org/fhir/organization-type"/>
+        <code value="Implementer"/> 
+    </coding>
+</type>
+<!--The contact object is repeated for each person listed in the directory-->
+<contact>
+    <purpose value="Operations"/>
+    <name>
+        <use value="official"/>
+        <text value="${contactName}"/>
+    </name>
+    <telecom>
+        <system value="email"/>
+        <value value="${email}"/>
+        <use value="work"/>
+    </telecom>
+    <telecom>
+        <system value="phone"/>
+        <value value="${phone}"/>
+        <use value="work"/>
+    </telecom>
+    <!--General address for this specific person-->
+    <address>
+        <use value="work"/>
+        <type value="both"/>
+        <line value="${addressLine1}"/>
+        <city value="${city}"/>
+        <postalCode value="${postalCode}"/>
+        <country value="USA"/>
+    </address>
+</contact>
+<!--General organizational physical and mailing address-->
+<address>
+    <use value="work"/>
+    <type value="both"/>
+    <line value="${addressLine1}"/>
+    <city value="${city}"/>
+    <state value="${state}"/>
+    <postalCode value="${postalCode}"/>
+    <country value="USA"/>
+    <extension url="OrgPosition">
+        <valueCodeableConcept>
+            <coding>
+                <system value="https://sequoiaproject.org/StructureDefinition/Address/Position/1.0.0"/>
+                <value>
+                    <position>
+                        <longitude value="${longitude}"/>
+                        <latitude value="${latitude}"/>
+                    </position>
+                </value>
+            </coding>
+        </valueCodeableConcept>
+    </extension>
+</address>
+<!--If this is a Carequality Connection, then the partOf object points to the Carequality Implementer-->
+<!--If this is a eHealth Exchange Sub-Participant, then the partOf object points to the eHealth Exchange Participant-->
+<!--If this is a lower-level organization, then the partOf object points to the immediate parent organization-->
+<!--If this is a top-level organization, then the partOf object is not specified-->
+<!--The Endpoints are represented as contained resources and are repeated as needed-->
+${endpointXCPD}
+${endpointDQ}
+${endpointDR}
+</Organization>`;
+}
+
+function getEndpoint(
+  oid: string,
+  url: string,
+  urlType: string,
+  channelType: string,
```

Comment:
Isn't there a 1:1 relationship between `urlType` and `channelType`?

Reply:
no, xca-dr and xca-dq both use ihe-xca channelType

Reply:
not sure what you mean here

Reply:
Can we call `getEndpoint` with these combinations of parameters?

- urlType="XCA ITI-39" + channelType: "ihe-xca"
- urlType="XCA ITI-39" + channelType: "ihe-xcpd"
- urlType="XCPD ITI-55" + channelType: "ihe-xca"
- urlType="XCPD ITI-55" + channelType: "ihe-xcpd"

Or can we only have a channel type "ihe-xcpd" when the url type is "XCPD ITI-55"?

If the latter, we can (and prob should) limit that on compile time, to avoid runtime errors. Do we even need the channel type in that case?

Reply:
Ah, I see. Yea, that's a good point